testsmainnet/itoken-collateral/test_itoken_collateral.py

```diff
@@ -0,0 +1,401 @@
+#!/usr/bin/python3
+
+import pytest
+from brownie import ZERO_ADDRESS, network, Contract, reverts, chain
+from brownie.convert.datatypes import Wei
+from eth_abi import encode_abi, is_encodable, encode_single
+from eth_abi.packed import encode_single_packed, encode_abi_packed
+from hexbytes import HexBytes
+import json
+from eth_account import Account
+from eth_account.messages import encode_structured_data
+from eip712.messages import EIP712Message, EIP712Type
+from brownie.network.account import LocalAccount
+from brownie.convert.datatypes import *
+from brownie import web3
+
+@pytest.fixture(scope="module")
+def requireFork():
+    assert (network.show_active() == "fork" or "fork" in network.show_active())
+
+
+@pytest.fixture(autouse=True)
+def isolation(fn_isolation):
+    pass
+
+
+@pytest.fixture(scope="module")
+def BZX(accounts, interface, TickMathV1, LoanOpenings, LoanSettings, ProtocolSettings, LoanClosingsLiquidation, LoanMaintenance, LiquidationHelper):
+    tickMathV1 = accounts[0].deploy(TickMathV1)
+    liquidationHelper = accounts[0].deploy(LiquidationHelper)
+
+    lo = accounts[0].deploy(LoanOpenings)
+    ls = accounts[0].deploy(LoanSettings)
+    ps = accounts[0].deploy(ProtocolSettings)
+    lcs = accounts[0].deploy(LoanClosingsLiquidation)
+    lm = accounts[0].deploy(LoanMaintenance)
+
+    bzx = Contract.from_abi("bzx", address="0xD8Ee69652E4e4838f2531732a46d1f7F584F0b7f", abi=interface.IBZx.abi)
+    bzx.replaceContract(lo, {"from": bzx.owner()})
+    bzx.replaceContract(ls, {"from": bzx.owner()})
+    bzx.replaceContract(ps, {"from": bzx.owner()})
+    bzx.replaceContract(lcs, {"from": bzx.owner()})
+    bzx.replaceContract(lm, {"from": bzx.owner()})
+
+    return bzx
+
+
+@pytest.fixture(scope="module")
+def USDT(accounts, TestToken):
+    return Contract.from_abi("USDT", address="0xdAC17F958D2ee523a2206206994597C13D831ec7", abi=TestToken.abi)
+
+@pytest.fixture(scope="module")
+def USDC(accounts, TestToken):
+    return Contract.from_abi("USDC", address="0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48", abi=TestToken.abi)
+
+@pytest.fixture(scope="module")
+def FRAX(accounts, TestToken):
+    return Contract.from_abi("FRAX", address="0x853d955aCEf822Db058eb8505911ED77F175b99e", abi=TestToken.abi)
+
+@pytest.fixture(scope="module")
+def AAVE(accounts, TestToken):
+    return Contract.from_abi("AAVE", address="0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9", abi=TestToken.abi)
+
+@pytest.fixture(scope="module")
+def STETH(accounts, TestToken):
```

Comment:
minor note but we plan to list wstETH, not stETH as stETH is rebasing. This isnt even used for the test case so not an issue

Reply:
I was just testing ability to add third party token as collateral only